skymarshal/skycmd/flags.go

```diff
@@ -242,3 +242,61 @@ func (con *Connector) newTeamConfig() (TeamConfig, error) {
 
 	return res, nil
 }
+
+type skyDisplayUserIdGenerator struct {
+	mapConnectorUserid map[string]string
+}
+
+func NewSkyDisplayUserIdGenerator(config map[string]string) (atc.DisplayUserIdGenerator, error) {
+	generator := &skyDisplayUserIdGenerator{
+		mapConnectorUserid: map[string]string{},
+	}
+
+	for connectorId, fieldName := range config {
+		valid := false
+		if connectorId == "local" {
+			valid = true
+		} else {
+			for _, connector := range GetConnectors() {
+				if connectorId == connector.ID() {
+					valid = true
+					break
+				}
+			}
+		}
+		if !valid {
+			return nil, fmt.Errorf("invalid connector: %s", connectorId)
+		}
+
+		if fieldName != "user_id" && fieldName != "name" && fieldName != "username" && fieldName != "email" {
```

Comment:
nit: This might be easier to skim as a `switch`:

```go
switch fieldName {
case "user_id", "name", "username", "email":
default:
    return nil, fmt.Errorf(...)
}
```

Reply:
Thanks for the suggestion. Fixed.